push

```diff
@@ -54,31 +64,63 @@ def _controlled_tempdir():
     finally:
         shutil.rmtree(td)
 
-def _build_fw(zip_path, apps_path):
-    with ZipFile(zip_path, 'w') as zf:
-        for fname in os.listdir(apps_path):
-            zf.write(os.path.join(apps_path, fname), fname)
-
-def _transfer_firmware(host, repo_path, scp, ssh, sensors):
+def _build_fw(zip_path, apps_path, targets):
+    if targets:
+        regex_list = [re.compile(f"{target}" + r"(.*).hex") for target in targets]
```

Comment:
The rear panel produces `.bin` files, you'll need to include those as well.

Reply:
We should define a list of targets and validate them against the user input, something like this.

```
TARGETS = [
    "head",
    "gantry-x",
    "gantry-y",
    "gripper",
    "pipettes-single",
    "pipettes-multi",
    "pipettes-96",
    "hepa-uv",
    "rear-panel",
]

for target in args.targets:
    if target not in TARGETS:
        print(f"Unknown target {target}", file=sys.stderr)
        continue

    # do work here
```

We actually do something similar in the `scripts/subsystem_versions.py` script